fnConsensus/types.go

```diff
@@ -51,6 +51,7 @@ type ReactorState struct {
 	PreviousTimedOutVoteSets map[string]*FnVoteSet // TODO: unused, consider removing
 	PreviousMajVoteSets      map[string]*FnVoteSet
 	PreviousValidatorSet     *types.ValidatorSet
+	Message                  map[string][]byte
```

Comment:
`Message` -> `Messages`
And lets store the hash with each message too, that way we can double-check the hash matches before calling `safeSubmitMultiSignedMessage`.